controllers/twophase/types.go

```diff
@@ -65,6 +68,47 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	}
 	chaos := _chaos.(v1alpha1.InnerSchedulerObject)
 
+	if chaos.GetStatus().Experiment.Phase == v1alpha1.ExperimentPhaseFinished {
+		// This chaos was finished, we should remove the finalizer
+		r.Log.Info("Removing pre-finalizer")
+
+		chaos.GetMeta().SetFinalizers(finalizer.RemoveFromFinalizer(chaos.GetMeta().GetFinalizers(), common.PreFinalizer))
+
+		updateError := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I think `RetryOnConflict` is unnecessary here, we can just return the error and wait for the next reconciling.